app/services/remove_status_service.rb

```diff
@@ -57,14 +53,37 @@ def remove_reblogs
     end
   end
 
-  def unpush(type, receiver, status)
-    if status.reblog? && !redis.zscore(FeedManager.instance.key(type, receiver.id), status.reblog_of_id).nil?
-      redis.zadd(FeedManager.instance.key(type, receiver.id), status.reblog_of_id, status.reblog_of_id)
+  def unpush_bulk(type, receiver_ids, status)
```

Comment:
This method needs a bit of changing, I think. It's hard to tell what is happening, and instead of one loop, it loops 3 times over the same array in the worst case, and 2 times in the best case.

It could help to split the method into two, one for when the status is a reblog, and one for when it's not. When it's not a reblog, the method is really simple: just loop over receiver_ids in a pipeline, zremrangebyscore and publish at the same time. When it's a reblog, that's the more complicated case, and I accept its solution, except I would prefer not to use lambdas in this new method.